clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -743,8 +743,11 @@ private String inferAssumedRoleArn(AmazonCredentials credentials) {
           "The given kind of credentials is not supported, "
               + "please report this issue to the Spinnaker project on Github.");
     }
-
-    return String.format("arn:aws:iam::%s:%s", credentials.getAccountId(), role);
+    if (!role.startsWith("arn:")) {
+      return String.format("arn:aws:iam::%s:%s", credentials.getAccountId(), role);
+    } else {
```

Comment:
since we're returning in the previous `if` statement, we don't need this `else` block. We can just do:
```
...
if (!role.startsWith("arn:")) {
      return String.format("arn:aws:iam::%s:%s", credentials.getAccountId(), role);
}
return role;
...
```